ntoskrnl/ke/amd64/trap.S

```diff
@@ -1150,21 +1150,27 @@ PUBLIC KiInitiateUserApc
     mov rax, APC_LEVEL
     mov cr8, rax
 
+    /* Get the current thread */
+    mov rbp, gs:[PcCurrentThread]
+
+deliver_apcs:
```

Comment:
I was comparing with the equivalent in x86, which appears to be:
https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/include/internal/i386/ke.h;hb=f92972efcabafe1552360f8750883db3de28df9a#l825
(there is also another function that does a similar job:
https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/include/internal/i386/trap_x.h;hb=f92972efcabafe1552360f8750883db3de28df9a#l322 )
and it turns out that instead, the APC_LEVEL raising/lowering is done within the loop, instead of outside as you do it here.

Reply:
That doesn't make a difference. as soon as interrupts are disabled no new APCs can be queued.